navigator/backend/src/test/scala/com/digitalasset/navigator/backend/query/ContractSortSpec.scala

```diff
@@ -81,13 +89,18 @@ class ContractSortSpec extends FlatSpec with Matchers {
     }
   }
 
+  implicit val sortParties: Ordering[Seq[String]] =
+    Ordering.fromLessThan[Seq[String]](_.mkString < _.mkString)
+
   test(List(), contracts)
   test(List("id" -> ASCENDING), contracts.sortBy(_.id.unwrap))
   test(List("id" -> DESCENDING), contracts.sortBy(_.id.unwrap)(Ordering[String].reverse))
   test(
     List("agreementText" -> ASCENDING),
     contracts.sortBy(_.agreementText)
   )
+  test(List("signatories" -> ASCENDING), contracts.sortBy(_.signatories))
```

Comment:
I'm surprised this test passes. You should need to edit [navigator/query/project/package.scala](https://github.com/digital-asset/daml/blob/master/navigator/backend/src/main/scala/com/digitalasset/navigator/query/project/package.scala#L204) in order to support sorting, which I don't see in this PR.

Looking at the code, sorting by `agreementText` should not work either.

The reason this test passes is probably that the local `test` method uses `theSameElementsAs` instead of `theSameElementsInOrderAs`, which makes this whole test suite useless.

Reply:
Very good catch, thanks! I fixed the test and this unveiled some apparently defective behavior in the existing code (in particular with regards to the agreement text and the argument), I'll include a fix as part of this PR.

Reply:
### Update

Implementing the projection for agreement text was trivial.

For arguments, I suspected the issue was rather that the test cases are formulated in a way that requires stable sorting, which is thought could not necessarily the case. After checking the Scala documentation (for version 2.12.8), [the underlying sorting algorithm is stated to be stable](https://www.scala-lang.org/api/2.12.8/scala/collection/Seq.html#sortBy[B](f:A=%3EB)(implicitord:scala.math.Ordering[B]):Repr), which is a guarantee I believe we can rely on.

I then noticed that we are relying on the default ordering of `Option`s when comparing values that may not be compatible with the sorting criterion (e.g. sorting by a field in the argument which is not present in certain contracts). The default ordering which puts `None`s before `Some`s, as in the following example:
```scala
List(None, Some(4)).sorted
List(Some(4), None).sorted
// both evaluate to List(None, Some(4))
```

Certain test cases reasonably require the values that are compatible with a sorting criterion to appear before those who don't. Unfortunately simply reversing the ordering for missing values causes other test cases to fail. I'll try to better understand the issue and add a fix as mentioned.

Reply:
Unfortunately the fix doesn't seem to be trivial and I'd prefer this to not hold back this contribution. I'll make sense of the failing test cases, "fix" them to represent the current status and open a bug report about the current behavior.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/1814#discussion_r296397390